FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +30,34 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: self.url) { [weak self] result in
+			guard let _ = self else { return }
+
+			guard case .success = result else {
+				completion(.failure(Error.connectivity))
+				return
+			}
+
+			guard let (data, httpResponse) = try? result.get(), httpResponse.statusCode == 200 else {
+				completion(.failure(Error.invalidData))
+				return
+			}
+
+			do {
+				let decoder = JSONDecoder()
+				decoder.keyDecodingStrategy = .convertFromSnakeCase
+				decoder.dateDecodingStrategy = .secondsSince1970
+				let response = try decoder.decode(FeedImageResponse.self, from: data)
+				let feedImages = response.items.map {
+					FeedImage(id: $0.imageId,
+					          description: $0.imageDesc,
+					          location: $0.imageLoc,
+					          url: $0.imageUrl)
+				}
+				completion(.success(feedImages))
+			} catch {
+				completion(.failure(Error.invalidData))
+			}
```

Comment:
Instead of using a `try` inside a `do/catch` block, since this portion of code does not care about any errors thrown by `JSONDecoder`, it's possible to achieve the same result with an optional `try?`.
```suggestion
			let decoder = JSONDecoder()
			decoder.keyDecodingStrategy = .convertFromSnakeCase
			guard let response = try? decoder.decode(FeedImageResponse.self, from: data) else {
				return completion(.failure(Error.invalidData))
			}
			
			let feedImages = response.items.map {
				FeedImage(id: $0.image_id,
						  description: $0.image_desc,
						  location: $0.image_loc,
						  url: $0.image_url)
			}
			completion(.success(feedImages))
```